src/dataunion/DataUnion.ts

```diff
@@ -449,15 +474,15 @@ export class DataUnion {
         recipientAddress: EthereumAddress,
         signature: string,
         sendToMainnet: boolean = true,
-    ): Promise<TransactionResponse> {
+    ) {
         const duSidechain = await this.getContracts().getSidechainContract(this.contractAddress)
         return duSidechain.withdrawAllToSigned(memberAddress, recipientAddress, sendToMainnet, signature)
     }
 
     /**
      * Admin: set admin fee (between 0.0 and 1.0) for the data union
      */
-    async setAdminFee(newFeeFraction: number): Promise<TransactionReceipt> {
+    async setAdminFee(newFeeFraction: number) {
```

Comment:
The return type should still be `Promise<TransactionReceipt>`?

Reply:
ContractReceipt now, and it's in fact correctly inferred. Manually typing it to TransactionReceipt (a supertype) was in fact harmful.